lib/rules/no-template-children.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+module.exports = {
+    meta: {
+        docs: {
+            description: 'prevent accessing the immediate children of this.template',
+            category: 'LWC',
+            recommended: true,
+            url: docUrl('no-template-children'),
+        },
+
+        schema: [],
+    },
+
+    create(context) {
+        const disallowedProperties = ['firstElementChild', 'firstChild', 'children', 'childNodes'];
```

Comment:
I prefer a `Set`, it will also help with `disallowedProperties.includes`. Also, what do you think about declaring it globally in the module?

Reply:
Done!